v2/pink-sb/src/lib/ToggleButton.svelte

```diff
@@ -24,16 +29,57 @@
             if (next === undefined || Array.isArray(next)) return curr;
             active = next;
 
+            updateIndicatorPosition();
             return next;
         }
     });
 
+    const updateIndicatorPosition = () => {
+        if (!containerRef || !indicator) return;
+
+        const activeEl = containerRef.querySelector(`[data-id="${active}"]`);
+        if (!activeEl) return;
+
+        const containerRect = containerRef.getBoundingClientRect();
+        const activeRect = activeEl.getBoundingClientRect();
+
+        const left = activeRect.left - containerRect.left - 1;
+
+        if (isInitialPosition) {
+            indicator.style.transition = 'opacity 0.2s ease-out';
+        }
+
+        indicator.style.transform = `translateX(${left}px)`;
+        indicator.style.width = `${activeRect.width}px`;
+        indicator.style.opacity = '1';
+
+        if (isInitialPosition) {
+            indicator.offsetHeight;
+            indicator.style.transition = '';
+            isInitialPosition = false;
+        }
+    };
+
+    onMount(() => {
+        updateIndicatorPosition();
+    });
+
     $: value.set(active ?? undefined);
+    $: if (active) {
+        setTimeout(updateIndicatorPosition, 0);
```

Comment:
Why the timeout with a 0s delay? If we are waiting for a tick, can't we also use `requestAnimationFrame()`?